### PR TITLE
Worker version support

### PR DESCRIPTION
**Pull request by natabeck**
Signed-off-by: Natasha Beck <natasha.kravtsov@hp.com>

In order to support worker in cluster we need to have version for each worker.
In order to have correct execution all workers should have the same version.

### CODE REVIEW

**Comment by orius123** [comment]
Y?

**Comment by orius123** [comment]
version "N/A"??

**Comment by natabeck** [comment]
If not set otherwise it is N/A. Later WorkerVersionService will be added that will define the workers version.

File: score-api/src/main/resources/io/cloudslang/schema/score.xsd [hunk 1]
```diff
@@ -21,6 +21,7 @@
             <xsd:attribute type="xsd:boolean" name="externalDatabase" />
             <xsd:attribute type="xsd:boolean" name="ignoreEngineJobs" />
             <xsd:attribute type="xsd:boolean" name="registerPauseResumeService"/>
+            <xsd:attribute type="xsd:boolean" name="registerWorkerNodeService"/>
```
Comment (orius123) [inline score-api/src/main/resources/io/cloudslang/schema/score.xsd]: Where  do you use this flag and what for?

File: engine/node/score-node-api/src/main/java/io/cloudslang/engine/node/entities/WorkerNode.java [hunk 2]
```diff
@@ -244,52 +247,84 @@ public void setWorkerRecoveryVersion(String workerRecoveryVersion) {
...
-				.append(this.ackTime, that.ackTime)
-                .append(this.ackVersion, that.ackVersion)
-				.append(this.os, that.os)
-				.append(this.jvm, that.jvm)
-				.append(this.dotNetVersion, that.dotNetVersion)
-				.append(this.groups, that.groups)
-                .append(this.bulkNumber, that.bulkNumber)
-                .append(this.workerRecoveryVersion, that.workerRecoveryVersion)
-				.isEquals();
+		if (active != that.active) return false;
```
Comment (orius123) [inline engine/node/score-node-api/src/main/java/io/cloudslang/engine/node/entities/WorkerNode.java]: Like the EqualsBuilder better, is it a matter of taste, or  that you found an issue with that.

**Comment by orius123** [comment]
Still didn't get the purpose of this change.
Version of what? recovery? am i missing something?

**Comment by natabeck** [comment]
In OO use case we are implementing auto upgrade for RASes. This means that each worker knows its version and its engine version.

Once the engine version is greater than his – it performs auto upgrade.

File: score-api/src/main/resources/io/cloudslang/schema/score.xsd (hunk 1 above)
Comment (natabeck) [inline score-api/src/main/resources/io/cloudslang/schema/score.xsd]: We are extending the WorkerNodesService in OO and like in PauseResume case want to be able to load extended implementation

**Comment by orius123** [comment]
Why do you need this version thing in the db. Make your custom impl send it on up or something. Dont really think its score concern.

**Comment by orius123** [comment]
?

**Pull request merged**